src/node/createSetupServer.ts

```diff
@@ -39,6 +39,7 @@ export function createSetupServer(...interceptors: Interceptor[]) {
         )
     })
     const interceptor = new RequestInterceptor(interceptors)
+    let pendingRequests: { [requestId: string]: NodeJS.Timeout } = {}
```

Comment:
Perhaps we could consider using `Set` for storing the pending requests?

```js
const pendingRequests = new Set<string, NodeJS.Timeout>()

pendingRequests.has(mockedRequest.id)
pendingRequest.delete(mockedRequest.id)
```

Reply:
I also updated the service worker (it seems like `Map` is supported accross older browsers).
Fyi, `Set` only holds values, so I used a `Map` which hold key-value pairs.